hazelcast/src/main/java/com/hazelcast/internal/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -312,6 +309,11 @@ public void onFailure(Throwable t) {
         }
     }
 
+    private boolean shouldTriggerMasterToAssignPartitions() {
+        ClusterServiceImpl clusterService = node.getClusterService();
+        return partitionStateManager.isInitialized() && clusterService.isJoined() && node.getNodeExtension().isStartCompleted();
```

Comment:
Should not this be `!partitionStateManager.isInitialized() && ...`?

Reply:
good catch. fixed